File: src/ManaFoodProduct.Domain/Entities/BaseEntity.cs
namespace ManaFoodProduct.Domain.Entities;

public abstract class BaseEntity
{
    public Guid Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public Guid CreatedBy { get; set; }
    public DateTime UpdatedAt { get; set; }
    public Guid UpdatedBy { get; set; }
    public bool Deleted { get; set; }
}

File: ManaFoodPoduct/src/main/kotlin/br/com/manafood/manafoodpoduct/domain/common/BaseEntity.kt
package br.com.manafood.manafoodpoduct.domain.common

import java.time.LocalDateTime
import java.util.UUID

abstract class BaseEntity(
    open val id: UUID? = null,
    open val createdAt: LocalDateTime = LocalDateTime.now(),
    open val createdBy: UUID,
    open val updatedAt: LocalDateTime? = null,
    open val updatedBy: UUID? = null,
    open val deleted: Boolean = false
)
